metadata/Cargo.toml

```diff
@@ -21,9 +21,11 @@ ink_primitives = { version = "2.1.0", path = "../primitives/", default-features
 serde = { version = "1.0", default-features = false, features = ["derive", "alloc"] }
 serde_json = "1.0"
 derive_more = { version = "0.99", default-features = false, features = ["from"] }
+proc-macro2 = "1.0"
 scale-info = { version = "0.3", default-features = false, features = ["derive"] }
 semver = { version = "0.10.0", features = ["serde"] }
 url = { version = "2.1.1", features = ["serde"] }
+quote = "1.0"
```

Comment:
`proc-macro2` and `quote` (as well as `syn`) dependencies are normally reserved for proc macro crates and since they are not `no_std` compatible should be avoided for ink! core crates. Maybe there is still a good reason for them here but from what I see right now this should not be included.

Reply:
The idea is for `cargo-contract` to create an instance of `InkProjectExtension` [here](https://github.com/paritytech/cargo-contract/pull/62/files#diff-e4f600b805550e47676a5a537a3bf4b7R87) (which contains the additional metadata), and then use that instance to generate the code to place in `main.rs` which will construct an equivalent instance and pass it to `__ink_generate_metadata`.

Ways forward:

1. Continue with this approach but make sure `proc_macro2` and `quote` are for `std` feature only
2. This approach but just use our own trait in place of `ToTokens` and strings to generate code
3. Make all fields public and implement `ToTokens` on `cargo-contract` side
4. Scrap this approach entirely and load the contract as a DLL using https://docs.rs/libloading/0.6.2/li__ink_generate_metadata(extension)bloading/ or similar, and calling `__ink_generate_metadata(extension)`

Reply:
This has now all been removed. The additional metadata data structures have all been moved to `cargo-contract` https://github.com/paritytech/cargo-contract/pull/62